### PR TITLE
ValueMappings: support settings icons in field value mappings

### LINKED ISSUE #43939: Support Icons For Value Mapping
Add support for icons for field value mapping

### PR DESCRIPTION
**Pull request by nmarrs**
**What this PR does / why we need it**:
Add support for icons for field value mapping.

This will allow us to allow users to use value mapping feature with our icons available in the `ResourcePicker.ts` modal
Screen Shot 2022-01-11 at 4 14 10 PM

**Which issue(s) this PR fixes**:

Fixes #43939

### CODE REVIEW

File: packages/grafana-data/src/field/displayProcessor.ts [hunk 1]
```diff
@@ -137,7 +142,7 @@ export function getDisplayProcessor(options?: DisplayProcessorOptions): DisplayP
       percent = scaleResult.percent;
     }
 
-    return { text, numeric, prefix, suffix, color, percent };
+    return { text, numeric, prefix, suffix, color, icon, percent };
```
Comment (ryantxu) [inline packages/grafana-data/src/field/displayProcessor.ts:145]: Maybe:

```suggestion
    const disp = { text, numeric, prefix, suffix, color };
    if (icon != null) {
       disp.icon = icon;
    }
    if (percent != null) {
       disp.percent = percent;
    }
    return disp;
```

would be nicer not to have all the explicit undefined values defined in each object.

**Review by torkelo — COMMENTED** [review]
Is this a draft? 

Think it needs an update to the value mapping editor and a update to a devenv test dashboard with this new mapping case

**Comment by nmarrs** [comment]
This is first step while we work with design to get UX figured out / finalized. We can merge this separately as it shouldn't hurt anything (optional field config) while we work on UX side of things. We can also leave as a draft though if there is any worries with any potential adverse side affects

**Review by ryantxu — APPROVED** [review]
I think it is OK to commit before the UX is finalized for actually implementing the change in the UI

**Pull request merged**